worker/instancemutater/worker.go

```diff
@@ -174,9 +176,12 @@ type mutaterWorker struct {
 }
 
 func (w *mutaterWorker) loop() error {
+	var wg sync.WaitGroup
+	defer wg.Wait()
```

Comment:
The only problem with wg.Wait is that it will hang if we are supposed to be dying if our code has bugs in it. But i think you've wired through dying already, so its probably ok.